src/Component/App/App.js

```diff
@@ -7,47 +6,50 @@ import SelectedMovie from '../SelectedMovie/SelectedMovie.js';
 
 function App() {
   const [movies] = useState(movieData.movies);
-  const [selectedMovie, setSelectedMovie] = useState(null)
-  const [showMainMovie, setShowMainMovie] = useState(true)
-  const [showMovieCards, setMovieCards]= useState(true)
+  const [selectedMovie, setSelectedMovie] = useState(null);
+  const [showMainMovie, setShowMainMovie] = useState(true);
+  const [showMovieCards, setShowMovieCards] = useState(true);
 
   const randomIndex = Math.floor(Math.random() * movies.length);
-  const randomMovie = movies[randomIndex]
+  const randomMovie = movies[randomIndex];
     
-  function handleMovieSelection (id){
-    const movie = movies.find(movie => movie.id ===id)
+  function handleMovieSelection(id) {
+    console.log("handleMovieSelection called with ID:", id);
+    const movie = movies.find(movie => movie.id === id);
+    console.log("Selected movie:", movie);
     setSelectedMovie(movie);
-    setShowMainMovie(false)
-    setMovieCards(false)
+    setShowMainMovie(false);
+    setShowMovieCards(false);
   }
+
   function handleBackToSelection() {
     setSelectedMovie(null);
```

Comment:
Same here, instead of null, initialize the state with the appropriate data type.